src/main/java/org/arend/ext/core/expr/CoreReferenceExpression.java

```diff
@@ -1,8 +1,9 @@
 package org.arend.ext.core.expr;
 
 import org.arend.ext.core.context.CoreBinding;
+import org.jetbrains.annotations.Contract;
 import org.jetbrains.annotations.NotNull;
 
 public interface CoreReferenceExpression extends CoreExpression {
-  @NotNull CoreBinding getBinding();
+  @Contract(pure = true) @NotNull CoreBinding getBinding();
```

Comment:
What's the point of this?

Reply:
Tell IntelliJ's control flow analyzer that this function is pure, so the return value never changes.
Imagine this code:

```java
if (a.getBinding() instanceof A) {
  ((A) a.getBinding()).someMethodOfA();
}
```

IntelliJ will say there's a nullability issue because `getBinding()` returns nullable, even the `instanceof A` implies it's notnull. If we annotate it as pure, IntelliJ will see that the method always return the same result and won't show the error

Reply:
I'll continue to add more to help IntelliJ's CFA understand the codebase, and provide better code inspections & suggestions. I also highly recommend you to do so -- this is why I was so urgent on replacing the annotations library. There's extra convenience from our own library 😉

Reply:
We can add this contract to almost all methods. I don't like that only this one is annotated. This sort of implies that other such methods are not pure. Also, I don't want to annotate all of them.

Reply:
We can annotate simple ones, like `isExplicit` or some simple getters (that returns the field directly).

Reply:
All of them are "simple". They are just getters.

Reply:
> Also, I don't want to annotate all of them.

Why?

Reply:
Do you think it's a burden in maintainability?

Reply:
We can also add the annotation incrementally. When there's a warning to be addressed and it can be fixed by an annotation, we add it.